notebooks/2018-10-05-create-netcdf.ipynb

```diff
@@ -0,0 +1,368 @@
+    "\n",
+    "Metadata is Data about Data. It tells us the dimensions and datatypes of the data, as well as arbitrary attributes.\n",
+    "\n",
+    "Since the header is at the beginning of the file, and can change its length when new data is added or removed, it is very advisable to first create the header before starting to write the data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create a NetCDF file in Fortran\n",
+    "----\n",
+    "\n",
+    "We start out by creating a single 2-Dimensional field using Fortran.\n",
+    "\n",
+    "### netcdf module\n",
+    "\n",
+    "While there is an old-style Fortran 77 NetCDF interface, I strongly recommend using the Fortran 90 module:\n",
+    "\n",
+    "```\n",
+    "program create_netcdf\n",
+    "    use netcdf\n",
+    "    implicit none\n",
+    "end program create_netcdf\n",
+    "```\n",
+    "\n",
+    "This is the most basic structure. To compile this program, you need the `netcdf` library in the `LD_LIBRARY_PATH`, on `raijin` this can be easily achieved by running the `module load netcdf` command (or any of the different versions).\n",
+    "\n",
+    "Then you need to compile it with the compiler options `-lnetcdf -lnetcdff`:\n",
+    "\n",
+    "```\n",
+    "$ module load intel-fc/2018.3.222 netcdf/4.6.1\n",
+    "$ ifort -o create_netcdf -lnetcdf -lnetcdff create_netcdf.f90\n",
+    "```\n",
+    "\n",
+    "### netcdf Fortran 90 API\n",
+    "\n",
+    "The full API can be read here: https://www.unidata.ucar.edu/software/netcdf/netcdf-4/newdocs/netcdf-f90.html\n",
+    "\n",
+    "But the simple version is: Every instruction is a call to a **function** staring with `nf90_`, which returns an integer value as a status. Status `0` (or `NF90_NOERR`) means that the call was successful.\n",
+    "\n",
+    "I strongly suggest writing a little `check` routine, like this:\n",
+    "\n",
+    "```\n",
+    "subroutine check(status, operation)\n",
+    "    use netcdf\n",
+    "    implicit none\n",
+    "    integer, intent(in) :: status\n",
+    "    character(len=*), intent(in) :: operation\n",
+    "    if (status == NF90_NOERR) return\n",
+    "    print *, \"Error encountered during \", operation\n",
+    "    print *, nf90_strerror(status)\n",
+    "    STOP 1\n",
+    "end subroutine check\n",
+    "```\n",
+    "\n",
+    "Then, after every call to any `nf90_` routine, you can call this check to see what happened. If all went fine, the call returns immediately, but if there was an error, you get a human-readable error output. \n",
+    "\n",
+    "### create an empty file\n",
+    "\n",
+    "To create a file, we need a file handle, an integer variable, by convention `ncid`, which, similar to the `unit` of `read` and `write` commands, references the file we created:\n",
```

Comment:
May be worth pointing out opening an existing file works differently

Reply:
I added a note to the beginning of the document.